tools/funcslower.py

```diff
@@ -222,12 +285,51 @@ def args_str(event):
         return ""
     return str.join(" ", ["0x%x" % arg for arg in event.args[:args.arguments]])
 
+def print_stack(event):
+    user_stack = []
+    stack_traces = b.get_table("stacks")
+
+    if args.user_stack and event.user_stack_id > 0:
+        user_stack = stack_traces.walk(event.user_stack_id)
+
+    kernel_stack = []
+    if args.kernel_stack and event.kernel_stack_id > 0:
+        kernel_tmp = stack_traces.walk(event.kernel_stack_id)
+
+        # fix kernel stack
+        for addr in kernel_tmp:
+            kernel_stack.append(addr)
+        # the later IP checking
+        if event.kernel_ip:
+            kernel_stack.insert(0, event.kernel_ip)
```

Comment:
I am not sure whether kernel_ip here is really useful or not. It will give you ip inside the last frame. Most other fools do not print out this extra ip. What is your particular concern here?

Reply:
The only reason was to better match what the profile script was doing. Originally I was working off a somewhat old version of profile, which is why it's a little out of sync now. For now I just removed the line, if you think it is good to do I can also port over all the newer changes from the profile script.